apps/rush-lib/src/cli/actions/BaseRushAction.ts

```diff
@@ -112,11 +125,62 @@ export abstract class BaseRushAction extends BaseConfiglessRushAction {
     return super.onExecute();
   }
 
-  protected get eventHooksManager(): EventHooksManager {
-    if (!this._eventHooksManager) {
-      this._eventHooksManager = new EventHooksManager(this.rushConfiguration);
+  protected mergeProjectsWithVersionPolicy(
+    projectsParameters: CommandLineStringListParameter,
+    versionPoliciesParameters: CommandLineStringListParameter
+  ): string[] {
```

Comment:
Might be more efficient to just return a `RushConfigurationProject` array here.

Reply:
I'm going to leave it as-is for now. There's no need to have the entire project be passed around when using this method, especially since everything that uses it only ever needs strings.